src/Docker/Client/Api.hs

```diff
@@ -121,6 +121,10 @@ stopContainer t cid = requestUnit POST $ StopContainerEndpoint t cid
 
 -- | Blocks until a container with the given 'ContainerID' stops,
 -- then returns the exit code
+-- 
+-- __NOTE__: this endpoint will not return a response until the container
+-- has stopped. This function may therefore fail with a timeout error if
+-- the timeout is configured incorrectly in the HTTP manager.
```

Comment:
The timeout isn't necessarily "configured incorrectly", but simply configured to be shorter than the container's execution time, right?  It seems like that might be intended behavior from the HTTP manager configuration.